src/uu/ls/src/ls.rs

```diff
@@ -800,12 +800,24 @@ impl Config {
                     // Read number as octal
                     match u16::from_str_radix(x, 8) {
                         Ok(v) => v,
-                        Err(_) => return Err(LsError::InvalidLineWidth(x.into()).into()),
+                        Err(err) => {
```

Comment:
what about moving this into a function like:

fn parse_width(x: &str, is_octal: bool) -> Result<u16, LsError> {
    let parse_result = if x.starts_with('0') && x.len() > 1 {
        u16::from_str_radix(x, 8)
    } else {
        x.parse::<u16>()
    };

    match parse_result {
        Ok(v) => Ok(v),
        Err(err) => match err {
            ParseIntError::Overflow => {
                eprintln!("The width value provided is too large. It has been set to the maximum possible value.");
                Ok(u16::MAX)
            },
            _ => Err(LsError::InvalidLineWidth(x.into()).into()),
        }
    }
}

Reply:
Yeah, I though what I did looked pretty ugly. This should look much cleaner now.